### PR TITLE
APPS-5: Fixed SprykMainForm for Spryks with arguments, fixed process run for docker environment.

### PR DESCRIPTION
**Pull request by pushokwhite**
- Developer(s): @pushokwhite

- Ticket: https://spryker.atlassian.net/browse/APPS-5

- Release Group: https://release.spryker.com/release-groups/view/3531

- PR Overview: https://release.spryker.com/release/pull-request-sdk/SprykGui/36

- merge: squash

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   SprykGui               | minor (currently 0.x)  |                       |

-----------------------------------------

##### Change log

Breaking Changes

- Increased Spryk module version dependency.
- Increased Symfony module version dependency.
- Adjusted `BuildController::indexAction()` to show the `PreBuildForm` instead of `SprykMainForm`.

Improvements

- Introduced `PreBuildForm` to choose the values for `BuildController::buildAction()`.
- Adjusted `SprykDetailsForm` so now it displays all the required arguments of the used Spryk.
- Introduced `BuildController::buildAction()` to show the `SprykMainForm`.
- Introduced `SprykDefinition.enterModuleManually` transfer property.
- Adjusted `SprykFacadeInterface::getSprykDefinitions()` dependency.
- Introduced `SprykGuiConfig::getTopLevelSprykCategoryName()`, `SprykGuiConfig::getCommonSprykCategoryName()` and `SprykGuiConfig::getSprykCategories()`.

Fixes

- Fixed process run for Docker SDK driven environment.
- Fixed typo in `SprykGuiFacadeInterface::getAccessibleTransfers()` method return type description.

### CODE REVIEW

File: src/SprykerSdk/Zed/SprykGui/Communication/Form/BaseSprykForm.php [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+     *
+     * @return mixed[]
+     */
+    protected function getRelevantArguments(array $arguments): array
+    {
+        $filteredArguments = [];
+
+        foreach ($arguments as $argumentName => $argumentDefinition) {
+            if (
+                in_array($argumentName, ['module', 'dependentModule', 'organization'])
```
Comment (dereuromark) [inline src/SprykerSdk/Zed/SprykGui/Communication/Form/BaseSprykForm.php:35]: Can we please use true for 3rd arg for stringish comparisons?

Comment (pushokwhite) [reply src/SprykerSdk/Zed/SprykGui/Communication/Form/BaseSprykForm.php]: Sure, fixed.

**Comment by matweew** [comment]
`"spryker-sdk/spryk": "^0.2.6 || 0.3.0",` new version should be supported

File: src/SprykerSdk/Zed/SprykGui/Communication/Form/PreBuildForm.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     *
+     * @return void
+     */
+    protected function addEnterModuleManuallyField(FormBuilderInterface $builder): void
+    {
+        $builder->add(static::FIELD_ENTER_MODULE_MANUALLY, ChoiceType::class, [
+            'choices' => [
+                'Choose existing' => false,
+                'Type module name' => true,
```
Comment (matweew) [inline src/SprykerSdk/Zed/SprykGui/Communication/Form/PreBuildForm.php:80]:
```suggestion
                'Enter name manually' => true,
```

Comment (pushokwhite) [reply src/SprykerSdk/Zed/SprykGui/Communication/Form/PreBuildForm.php:80]: Fixed

File: src/SprykerSdk/Zed/SprykGui/Business/Spryk/Spryk.php [hunk 3]
```diff
@@ -144,25 +165,29 @@ protected function organizeSprykDefinitions(array $sprykDefinitions): array
...
     }
 
     /**
      * @param string $sprykName
+     * @param mixed[] $sprykDefinition
      *
      * @return string
      */
-    protected function getApplicationBySprykName(string $sprykName): string
+    protected function getApplication(string $sprykName, array $sprykDefinition): string
```
[the remark this answers is not in the document]
Comment (pushokwhite) [reply src/SprykerSdk/Zed/SprykGui/Business/Spryk/Spryk.php:177]: Fixed

File: src/SprykerSdk/Zed/SprykGui/SprykGuiConfig.php [hunk 4]
```diff
@@ -43,4 +46,43 @@ public function isSprykWebInterfaceEnabled(): bool
...
+
+    /**
+     * @api
+     *
+     * @return string[]
+     */
+    public function getApplicationNames(): array
+    {
+        return [
+            static::SPRYK_APPLICATION_NAME_TOP_LEVEL,
```
[the remark this answers is not in the document]
Comment (pushokwhite) [reply src/SprykerSdk/Zed/SprykGui/SprykGuiConfig.php:78]: Fixed

File: composer.json [hunk 5]
```diff
@@ -6,7 +6,7 @@
     "require": {
         "php": ">=7.3",
         "roave/better-reflection": "^4.3.0",
-        "spryker-sdk/spryk": "^0.2.6",
+        "spryker-sdk/spryk": "^0.3.0",
```
[the remark this answers is not in the document]
Comment (pushokwhite) [reply composer.json:9]: Checked the versions:
https://github.com/spryker/symfony/blob/3.5.0/composer.json contains `"symfony/console": "^3.4.0 || ^4.0.0 || ^5.0.0",`
https://github.com/symfony/console/blob/v3.4.0/composer.json contains `"symfony/process": "~3.3|~4.0",`
https://github.com/symfony/process/blob/3.3/Process.php is the first version where an array becomes an option for the first param.

File: src/SprykerSdk/Zed/SprykGui/Business/Spryk/Spryk.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (matweew) [reply src/SprykerSdk/Zed/SprykGui/Business/Spryk/Spryk.php]: $application = $this->getSprykCategories($sprykName, $sprykDefinition);

why plural and variable name is incorrectct

[the remark this answers is not in the document]
Comment (pushokwhite) [reply src/SprykerSdk/Zed/SprykGui/Business/Spryk/Spryk.php]: Fixed

**Review by matweew — APPROVED** [review]

**Review by matweew — APPROVED** [review]

**Review by matweew — APPROVED** [review]

**Comment by dereuromark** [comment]
There is no CI running on this?

**Pull request merged**